packages/agreements-service/src/controllers/mandatesController.ts

```diff
@@ -64,23 +119,35 @@ export async function store (ctx: AppContext): Promise<void> {
   }
 }
 
-export async function update (ctx: AppContext): Promise<void> {
-  logger.debug('Update mandate request', { body: ctx.request.body, headers: ctx.request.headers })
+export async function update(ctx: AppContext): Promise<void> {
+  logger.debug('Update mandate request', {
+    body: ctx.request.body,
+    headers: ctx.request.headers
+  })
 
   try {
-    const mandate = await Agreement.query().where('id', ctx.request.params['id']).andWhere('type', 'mandate').first()
+    const mandate = await Agreement.query()
+      .where('id', ctx.request.params['id'])
+      .andWhere('type', 'mandate')
+      .first()
     if (!mandate) {
       ctx.response.status = 404
       ctx.response.message = 'No mandate found'
       return
     }
+    if (mandate.cancelled) {
+      ctx.response.status = 400
+      ctx.response.message = 'Cancelled mandate'
+      return
+    }
 
     // TODO validate user owns account
-    const { userId, accountId, scope } = ctx.request.body
+    const { userId, accountId, scope, cancelled } = ctx.request.body
```

Comment:
This logic shouldn't live here.

Better to have a `cancelAgreementsController` that handles this logic. This would map to a url of `/agreements/{id}/cancel` which takes a POST request with an empty body to cancel the agreement